module/mempool/stdmap/chunk_statuses.go

```diff
@@ -67,3 +78,25 @@ func (cs ChunkStatuses) All() []*verification.ChunkStatus {
 func (cs ChunkStatuses) Size() uint {
 	return cs.Backend.Size()
 }
+
+// storedChunkStatus is an internal type for storing ChunkStatus in the mempool.
+//
+// It is the same as ChunkStatus, but additionally it implements an Entity type which
+// makes it storable in the mempool.
+// Note that as an entity, the ID of a storedChunkStatus is computed as the ID of the chunk locator
+// it represents. However, the usage of ID method is only confined to maintaining it on the mempool.
+// That is the motivation behind making it an internal type to make sure that no further decision out of
+// this package is taken based on ID of storedChunkStatus.
+type storedChunkStatus struct {
+	ChunkIndex      uint64
+	BlockHeight     uint64
+	ExecutionResult *flow.ExecutionResult
+}
+
+func (s storedChunkStatus) ID() flow.Identifier {
+	return chunks.ChunkLocatorID(s.ExecutionResult.ID(), s.ChunkIndex)
```

Comment:
Currently is used once per storing into memory (`Add`) and also once per adjusting the size of `Backend` if it gets full. Not sure if caching worth it, but can address it in a separate PR if you recommend (so that it will be reversible if needed).